src/averageAges.js

```diff
@@ -55,7 +85,33 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const newArr = onlyWithSon
+    ? people.filter(person => person.sex === 'm')
+    : people;
+
+  const ages = newArr.map((person) => {
+    const mother = person.mother;
+
+    const findMother = item => {
```

Comment:
```suggestion
    const findMother = item => {
```
item?😅
think about better name